packages/flutter/lib/src/services/platform_views.dart

```diff
@@ -1172,15 +1172,20 @@ class TextureAndroidViewController extends AndroidViewController {
   Future<void> _sendCreateMessage({required Size size, Offset? position}) async {
     assert(!size.isEmpty, 'trying to create $TextureAndroidViewController without setting a valid size.');
 
-    _internals.textureId = await _AndroidViewControllerInternals.sendCreateMessage(
+    final dynamic response = await _AndroidViewControllerInternals.sendCreateMessage(
       viewId: viewId,
       viewType: _viewType,
       hybrid: false,
       layoutDirection: _layoutDirection,
       creationParams: _creationParams,
       size: size,
       position: position,
-    ) as int;
+    );
+    if (response is int) {
+      _internals.textureId = response;
+    } else {
+      assert(response == null, 'response will be null when running under test');
```

Comment:
Relaxing production code requirements to satisfy tests is kind of worrying. Why can't we do a dummy channel response in the unit tests?

Reply:
Here's a trivial example of that, if it helps: https://github.com/flutter/packages/blob/main/packages/url_launcher/url_launcher_linux/test/url_launcher_linux_test.dart#L17-L25

A more complex version could check the call details and return an appropriate fake response.